template/cordova/lib/deployment.js

```diff
@@ -0,0 +1,272 @@
+/** 
+ * Enumerates devices attached to the development machine.
+ * @returns A Promise for an array of objects, which should be passed into other functions to represent the device.
+ * @remarks The returned objects contain 'index', 'name', and 'type' properties indicating basic information about them, 
+ *    which is limited to the information provided by the system.  Other properties may also be included, but they are 
+ *    specific to the deployment tool which created them and are likely used internally.
+ */
+DeploymentTool.prototype.enumerateDevices = function() {
+    return Q.reject('May not use DeploymentTool directly, instead get an instance from DeploymentTool.getDeploymentTool()');
+};
+
+/**
+ * Installs an app package to the target device.
+ * @returns A Promise which will be fulfilled on success or rejected on failure.
+ * @param pathToAppxPackage The path to the .appx package to install.
+ * @param targetDevice An object returned from a successful call to enumerateDevices.
+ * @shouldLaunch Indicates whether to launch the app after installing it.
+ * @shouldUpdate Indicates whether to explicitly update the app, or install clean.
+ * @pin Optionally provided if the device requires pairing for deployment.
+ */
+DeploymentTool.prototype.installAppPackage = function(pathToAppxPackage, targetDevice, shouldLaunch, shouldUpdate, pin) {
+    return Q.reject('May not use DeploymentTool directly, instead get an instance from DeploymentTool.getDeploymentTool()');
+};
+
+/**
+ * Uninstalls an app package from the target device.
+ * @returns A Promise which will be fulfilled on success or rejected on failure.
+ * @param packageInfo The app package name or Phone GUID representing the app.
+ * @param targetDevice An object returned from a successful call to enumerateDevices.
+ */
+DeploymentTool.prototype.uninstallAppPackage = function(packageInfo, targetDevice) {
+    return Q.reject('Unable to uninstall any app packages because that feature is not supported.');
+};
+
+/**
+ * Gets a list of installed apps on the target device.  This function is not supported for 
+ * Windows Phone 8.1.
+ * @param targetDevice {Object} An object returned from a successful call to enumerateDevices.
+ * @returns A Promise for an array of app names.
+ */
+DeploymentTool.prototype.getInstalledApps = function(targetDevice) {
+    return Q.reject('Unable to get installed apps because that feature is not supported.');
+};
+
+/**
+ * Launches an app on the target device.  This function is not supported for Windows 10.
+ * @param packageInfo {String} The app package name or Phone GUID representing the app.
+ * @param targetDevice {Object} An object returned from a successful call to enumerateDevices.
+ * @returns A Promise for when the app is launched.
+ */
+DeploymentTool.prototype.launchApp = function(packageInfo, targetDevice) {
+    return Q.reject('Unable to launch an app because that feature is not supported.');
+};
+
+/**
+ * Gets a DeploymentTool for use in 
```

Comment:
You were so excited to nit the trailing whitespace you didn't note that the comment was incomplete.